### DISCUSSION TITLE
Get the mouse position when onSelectionChange

### DISCUSSION THREAD

**Discussion by alanhe421** [sourcelair/xterm.js#5152] [Q&A] [answered]
Can I get the mouse position when the onSelectionChange event triggers? 
When the user selects a piece of text in the terminal, I need to display a UI toolbar at the right position. I noticed that the current callback method for the onSelectionChange event does not provide any parameters about the event.

```js
term.onSelectionChange(() => {
  
  term.getSelection() && copyToClipboard(term.getSelection());
  console.log('onSelectionChange', term.getSelection());

})
```

## version info

xterm:5.3.0

**Comment by jerch** [comment, marked as answer]
No, thats currently not exposed by the selection logic. It also never got asked before, so this seems to be of limited usage/demand.

What you can do as a workaround - in `onSelectionChange` grab the end position of the selection and calc the offset yourself based on the text metrics. Thats also more accurate than the mouse position (mouse might be at half the line/char). With the text metrics you can perfectly align your popup to the text cells.

**Reply by alanhe421** [reply]
thank you.
